frontend/packages/citizen-frontend/src/messages/MessagesPage.tsx

```diff
@@ -0,0 +1,59 @@
+import React, { useEffect, useState } from 'react'
+import Container from '@evaka/lib-components/src/layout/Container'
+import { Gap } from '@evaka/lib-components/src/white-space'
+import MessagesList from '~messages/MessagesList'
+import MessageReadView from '~messages/MessageReadView'
+import { Loading, Result, Success } from '@evaka/lib-common/src/api'
+import { ReceivedBulletin } from '~messages/types'
+import { useRestApi } from '@evaka/lib-common/src/utils/useRestApi'
+import { getBulletins, markBulletinRead } from '~messages/api'
+import AdaptiveFlex from '@evaka/lib-components/src/layout/AdaptiveFlex'
+import { messagesBreakpoint } from '~messages/const'
+
+function MessagesPage() {
+  const [bulletins, setBulletins] = useState<Result<ReceivedBulletin[]>>(
+    Loading.of()
+  )
+  const [activeBulletin, setActiveBulletin] = useState<ReceivedBulletin | null>(
+    null
+  )
+
+  const loadBulletins = useRestApi(getBulletins, setBulletins)
+  useEffect(() => loadBulletins(), [])
+
+  const openBulletin = (bulletin: ReceivedBulletin) => {
+    markBulletinRead(bulletin.id)
+
+    setActiveBulletin({
+      ...bulletin,
+      isRead: true
+    })
+
+    if (bulletins.isSuccess) {
+      setBulletins(
```

Comment:
I would do this state update only after the `markBulletinRead` request succeeds.